### DISCUSSION TITLE
no grace period when sending email notifications

### DISCUSSION THREAD

**Discussion by traktuner** [immich-app/immich#10508] [Feature Request]
### The bug

We have some shared albums across the family.
When I add 10 pictures in one minute one by one, me and my family members each get 10 email notifications.
Since I use a 3rd party service for those automated emails, the number of emails which can be sent in 24 hours is limited.
It would be great if just one notification per shared album can be sent out if there were no pictures added for like 5 minutes.
If someone else adds pictures to that shared album, another "5 minute timer" should start and then send another notification (since I also would like to know if someone else adds pictures to that album).
... 5 minutes just as an example ...

Also, it would be cool if it can be mentioned in the email notification who added pictures to that album (but that would be a feature request).

Thank you!

### The OS that Immich Server is running on

not relevant

### Version of Immich Server

1.106.4

### Version of Immich Mobile App

1.106.4

### Platform with the issue

- [ ] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
not relevant
```

### Your .env content

```Shell
not relevant
```

### Reproduction steps

```bash
not relevant
```

**Comment by traktuner** [comment]
@mmomjian @alextran1502 
In my opinion, this is not a feature request. Spamming people (at least in my case) with emails is not normal behavior and I consider this as a bug.

**Reply by alextran1502** [reply]
Sure, we will fix it

**Comment by tobiasfrejo** [comment]
I strongly agree. I don't use an external email service, but got my domain rate limited by Google after adding ~150 images one by one to an album shared with 3 Gmail users

**Reply by traktuner** [reply]
maybe @alextran1502 or @bo0tzz can convert this back into an issue...
Someone converted my issue into a discussion, which is just wrong.

**Reply by mmomjian** [reply]
Currently the email function is working as intended. We hope to add this feature because we agree there is a need for it but it’s currently not supported yet. Issues are for when an existing function isn’t working.

**Reply by traktuner** [reply]
Thank you for your response @mmomjian, but I respectfully disagree that this behavior is "working as intended."

While it's true that you can't fully control how users interact with the app, it's crucial to take measures to prevent this misbehavior. For instance, if a user—or multiple users on a server—add hundreds of pictures to a shared album one by one, that's a legitimate use case. However, the current approach to sending notifications can overwhelm users with emails.

Consider a scenario where 10 people share a family album. If 9 of them receive 150 individual email notifications, that becomes quite overwhelming and could negatively impact the user experience. A potential solution might be to sending tose notifications every 5 minutes or so, to reduce the email load.

I interpreted @alextran1502  comment that this is definitely an issue.

And as a side note, I am using a family album shared with some people. And I got some questions about the email spam. Then I had to deactivate the otherwise great feature.

**Reply by mmomjian** [reply]
If Alex disagrees he’s welcome to convert it - obviously he has the final say. Otherwise we will keep it here for now.

**Reply by alextran1502** [reply]
We can convert it so we can be reminded to work on this

**Comment by bo0tzz** [comment]
#12424
